packages/notifier/src/storesub.js

```diff
@@ -2,9 +2,76 @@
 import { E } from '@endo/eventual-send';
 import { Far, makeMarshal } from '@endo/marshal';
 import { observeIteration } from './asyncIterableAdaptor.js';
+import { makePublishKit, subscribeEach } from './publish-kit.js';
 import { makeSubscriptionKit } from './subscriber.js';
 
 /**
+ * Begin iterating the source, storing serialized iteration values.  If the
+ * storageNode's `setValue` operation rejects, the iteration will be terminated.
+ *
+ * Returns a StoredSubscriber that can be used by a client to directly follow
+ * the iteration themselves, or obtain information to subscribe to the stored
+ * data out-of-band.
+ *
+ * @template T
+ * @param {Subscriber<T>} subscriber
+ * @param {ERef<StorageNode>} storageNode
+ * @param {ERef<ReturnType<typeof makeMarshal>>} marshaller
+ * @returns {StoredSubscriber<T>}
+ */
+export const makeStoredSubscriber = (subscriber, storageNode, marshaller) => {
+  assert(subscriber && storageNode && marshaller, 'missing argument');
+
+  // Abort the iteration on the next observation if the publisher ever fails.
+  let publishFailed = false;
+  let publishException;
+
+  const fail = err => {
+    publishFailed = true;
+    publishException = err;
+  };
+
+  // Must *not* be an async function, because it sometimes must throw to abort
+  // the iteration.
+  const publishValue = obj => {
+    if (publishFailed) {
+      // To properly abort the iteration, this must be a synchronous exception.
+      throw publishException;
+    }
+
+    // Publish the value, capturing any error.
+    E(marshaller)
+      .serialize(obj)
+      .then(serialized => {
+        const encoded = JSON.stringify(serialized);
+        return E(storageNode).setValue(encoded);
+      })
+      .catch(fail);
+  };
+
+  // Start publishing the source.
+  observeIteration(subscribeEach(subscriber), {
+    updateState: publishValue,
+    finish: publishValue,
+  }).catch(fail);
+
+  /** @type {Unserializer} */
+  const unserializer = Far('unserializer', {
+    unserialize: E(marshaller).unserialize,
+  });
+
+  /** @type {StoredSubscriber<T>} */
+  const storesub = Far('StoredSubscriber', {
+    ...subscriber,
+    getStoreKey: E(storageNode).getStoreKey,
+    getUnserializer: () => unserializer,
```

Comment:
What advantage does this indirection provide?
```suggestion
    unserialize: E(marshaller).unserialize,
```

Reply:
An `Unserializer` is an attenuated, read-only `Marshaller`. This is copied but I think the need was to pass around "Marshaller" or "read only Marshaller" to where it's needed. They have to be far objects. Example use:
https://github.com/Agoric/agoric-sdk/blob/b662e1995831973f996ccbfc823ccb4c7c896895/packages/casting/src/follower-cosmjs.js#L344

Since this is a read-only Marshaller, I do think it would be more consistent to have the names relate. Maybe `Unmarshaller`?

I think it would help to have this attenuation be done by a function to help document the pattern. @kriskowal WDYT of that and being  next to `makeMarshall` in `@endo/marshall`?

Reply:
I'm mostly wondering if there's value in the object returned from `makeStoredSubscriber` providing a dedicated getter for such an object vs. just itself being a superset of one—e.g., `const unserializerP = E(storedSubscriber).getUnserializer()` vs. `const unserializer = Far('unserializer', { unserialize: E(storedSubscriber).unserialize })` or (depending upon trust level) `const unserializer = storedSubscriber`. This is not a blocking concern, just food for thought.

Reply:
> value in the object returned from `makeStoredSubscriber` providing a dedicated getter for such an object

yes, I think the value is symmetry with `Marshaller`. We can refactor this later for MN-2.